test/unit/test_devices/test_eos_device.py

```diff
@@ -31,17 +31,17 @@ def test_config(self):
         result = self.device.config(command)
 
         self.assertIsNone(result)
-        self.device.native.config.assert_called_with([command])
+        self.device.native.config.assert_called_with(command)
 
     def test_bad_config(self):
         command = "asdf poknw"
 
-        with self.assertRaisesRegex(CommandError, command):
+        with self.assertRaisesRegex(CommandError, command[0]):
             self.device.config(command)
 
     def test_config_list(self):
         commands = ["interface Eth1", "no shutdown"]
-        result = self.device.config_list(commands)
```

Comment:
This test case should mock self.device.config method and just assert that using config_list method calls the config method and passing the same list of commands to config that it received. An example can be found here: https://github.com/networktocode/pyntc/blob/4e56dfcede757a393e0d1caaa44c46539d949362/test/unit/test_devices/test_base_netmiko.py#L175-L181

This example will not be exactly the same since this is still using older style test cases, and the mocking is done differently there. The logic should still be the same, just need to swap out the mock object and use a method instead of a function

The reason is that the current test case is testing the functionality of what is now in the config method, which is not the goal of unit tests.

Reply:
@jmcgill298 when you say mock self.device.config would that be a fixture in this case?  I think I'm getting confused that this class in the test file is using unittest and I'm trying to make the assertions with pytest

Reply:
This would not be a fixture since this was built using unittest, and has not been converted to our newer testing strategies. You can just use a decorator like

```
@mock.patch.object(EOSDevice, "config")
def test_config_list(self, mock_config):
    commands = ["interface Eth1", "no shutdown"]
    self.device.config_list(commands)
    self.device.config.assert_called_with(commands)
```